generate.py

```diff
@@ -73,7 +84,7 @@ def normalize(self, freqDistSpec):
         prob_for_max_freq = list(filter(lambda x: x[0] == max_freq, temp_normailized))[0][1]
 
         distButMax = list(filter(lambda x: x[0] != max_freq, temp_normailized))
-        implied_prob_for_max_freq = 1 - sum([prob for (val, prob) in distButMax])
+        implied_prob_for_max_freq = round(1 - sum([prob for (val, prob) in distButMax]), 3)
```

Comment:
will rounding break the following assert? I don't know. 

you may consider removing rounding, or do `math.ceil(x * 1000) / 1000` for safety.

Reply:
Thanks for the alternative but no, it doesn't break. Tested it.